pkg/flowkit/gateway/emulator.go

```diff
@@ -65,71 +81,47 @@ func newEmulator(serviceAccount *flowkit.Account) *emulator.Blockchain {
 }
 
 func (g *EmulatorGateway) GetAccount(address flow.Address) (*flow.Account, error) {
-	return g.emulator.GetAccount(address)
+	return g.backend.GetAccount(g.ctx, address)
 }
 
 func (g *EmulatorGateway) SendSignedTransaction(tx *flowkit.Transaction) (*flow.Transaction, error) {
-	t := tx.FlowTransaction()
-	err := g.emulator.AddTransaction(*t)
-	if err != nil {
-		return nil, fmt.Errorf("failed to submit transaction: %w", err)
-	}
-
-	_, err = g.emulator.ExecuteNextTransaction()
-	if err != nil {
-		return nil, fmt.Errorf("failed to submit transaction: %w", err)
-	}
-
-	_, err = g.emulator.CommitBlock()
-	if err != nil {
-		return nil, fmt.Errorf("failed to submit transaction: %w", err)
-	}
-
-	return t, nil
+	return tx.FlowTransaction(), g.backend.SendTransaction(context.Background(), *tx.FlowTransaction())
```

Comment:
I think it's good to be explicit in returning values:
```go
err := g.backend.SendTransaction(context.Background(), *tx.FlowTransaction())
if err != nil {
   return nil, err
}

return t, nil
```
Just because if error exists the developer might not check for it and just use the returned value (in this case it's returned regardless) which can lead to errors. I remember reading that somewhere as a good practice but not sure where. This is a bit nit tho.

Reply:
This is really good advice actually.